django/db/models/sql/query.py

```diff
@@ -571,9 +565,9 @@ def combine(self, rhs, connector):
         self.where.add(w, connector)
 
         # Selection columns and extra extensions are those provided by 'rhs'.
-        self.select = []
-        for col in rhs.select:
-            self.add_select(col.relabeled_clone(change_map))
+        cols = [col.relabeled_clone(change_map) for col in rhs.select]
+        if cols:
```

Comment:
To match the previous code there should be a `self.select = ()` above this line. There's probably a bug being introduced here that's not tested, I think from combining two querysets where the LHS has set select whilst the RHS hasn't